pkg/kotsadm/operator.go

```diff
@@ -1,274 +0,0 @@
-	}
-	var deployment bytes.Buffer
-	if err := s.Encode(operatorDeployment, &deployment); err != nil {
-		return nil, errors.Wrap(err, "failed to marshal operator deployment")
-	}
-	docs["operator-deployment.yaml"] = deployment.Bytes()
-
-	return docs, nil
-}
-
-func ensureOperator(deployOptions types.DeployOptions, clientset *kubernetes.Clientset) error {
-	if deployOptions.EnsureRBAC {
-		if err := ensureOperatorRBAC(deployOptions, clientset); err != nil {
-			return errors.Wrap(err, "failed to ensure operator rbac")
-		}
-	}
-
-	if err := ensureOperatorDeployment(deployOptions, clientset); err != nil {
-		return errors.Wrap(err, "failed to ensure operator deployment")
-	}
-
-	return nil
-}
-
-func ensureOperatorClusterRBAC(deployOptions types.DeployOptions, clientset *kubernetes.Clientset) error {
-	err := ensureOperatorClusterRole(clientset)
-	if err != nil {
-		return errors.Wrap(err, "failed to ensure operator cluster role")
-	}
-
-	if err := ensureOperatorClusterRoleBinding(deployOptions.Namespace, clientset); err != nil {
-		return errors.Wrap(err, "failed to ensure operator cluster role binding")
-	}
-
-	if err := ensureOperatorServiceAccount(deployOptions.Namespace, clientset); err != nil {
-		return errors.Wrap(err, "failed to ensure operator service account")
-	}
-
-	return nil
-}
-
-func ensureOperatorRBAC(deployOptions types.DeployOptions, clientset *kubernetes.Clientset) error {
-	shouldBeClusterScoped, err := isOperatorClusterScoped(deployOptions.ApplicationMetadata)
-	if err != nil {
-		return errors.Wrap(err, "failed to check if operator should be cluster scoped")
-	}
-
-	// if this is cluster scoped, it's easy... create everything as a cluster role and cluster role binding
-	// with pretty open permissions
-	if shouldBeClusterScoped {
-		return ensureOperatorClusterRBAC(deployOptions, clientset)
-	}
-
-	// we want to ensure that the principle of least privilege is applied.
-	// so we will create our role and rolebinding
-	// and then create a role and role binding PER namespace that the application
-	// wants...  everthing will be linked to the same service account
-
-	err = ensureOperatorRole(deployOptions.Namespace, clientset)
-	if err != nil {
-		return errors.Wrap(err, "failed to ensure operator role")
-	}
-
-	if err := ensureOperatorRoleBinding(deployOptions.Namespace, deployOptions.Namespace, clientset); err != nil {
-		return errors.Wrap(err, "failed to ensure operator role binding")
-	}
-
-	decode := scheme.Codecs.UniversalDeserializer().Decode
-	obj, gvk, err := decode(deployOptions.ApplicationMetadata, nil, nil)
-	if err != nil {
-		return errors.Wrap(err, "failed to decode application metadata")
-	}
-
-	if gvk.Group != "kots.io" || gvk.Version != "v1beta1" || gvk.Kind != "Application" {
-		return errors.New("application metadata contained unepxected gvk")
-	}
-
-	application := obj.(*kotsv1beta1.Application)
-	for _, additionalNamespace := range application.Spec.AdditionalNamespaces {
```

Comment:
I don't see where we crate kotsadm RBAC in all application namespaces

Reply:
ah nice catch, i bet that's why the multi-namespaces test is failing